test/framework/log/collector.go

```diff
@@ -62,7 +66,18 @@ func (collector MachineLogCollector) CollectMachineLog(_ context.Context, _ clie
 				return err
 			}
 			defer f.Close()
-			return executeRemoteCommand(f, hostIPAddr, command, args...)
+			var lastErr error
+			// Try with all available IPs unless it succeeded.
+			for _, machineIPAddress := range machineIPAddresses {
+				lastErr = executeRemoteCommand(f, machineIPAddress, command, args...)
```

Comment:
Let's collect/append the errors and then return the aggregation if len > 0 in l.78

Reply:
(would be good if we include the actual error messages in the returned error)

Reply:
adjusted, PTAL

Reply:
Thx!